test/libsolidity/SolidityParser.cpp

```diff
@@ -113,6 +113,14 @@ while(0)
 
 BOOST_AUTO_TEST_SUITE(SolidityParser)
 
+BOOST_AUTO_TEST_CASE(unsatisfied_version)
+{
+	char const* text = R"(
+		pragma solidity ^99.99.0;
+	)";
+	CHECK_PARSE_ERROR(text, "Source file requires different compiler version");
+}
+
```

Comment:
Please add another test that shows that for source with syntax errors, we get an error for the pragma (the whole point of this PR ;) ).